### ISSUE TITLE
[2016 NIPS] Understanding the Effective Receptive Field in Deep Convolutional Neural Networks

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#37]
해당 논문은 theoretical receptive field와 effective receptive field가 다름을 이론과 실험을 통해 밝힘

논문의 핵심 내용인 effective receptive field와 관련 실험들, 극복할 방법 등만 설명

### 1.1. Intuition
Receptive field의 center pixels는 다양한 path로 output에 영향을 주며 gradient magnitude가 클 수 있음

그에 반해 outer area pixels는 path의 수 자체가 적어 output에 영향을 주기 힘들어 gradient magnitude가 작을 수 있음

이를 통해 receptive field의 모든 pixel이 output에 똑같이 영향을 주기보다는 center pixel이 output에 더 많은 영향을 준다고 말할 수 있음
(not all pixels in a receptive field contribute equally to an output unit's response, instead pixels at the center of a receptive field have a much larger impact on an output)

이는 이론적으로 network가 deep해질 수록 receptive field는 선형적으로 증가한다는 것과 반대되는 개념임

저자들은 가정을 통한 이론적 증명과 실험을 통해 실제 receptive field인 effective receptive field는 Gaussian distribution을 따른다는 것을 보임

### 1.2. ERFs are Gaussian distributed
![image](https://user-images.githubusercontent.com/49643709/136804155-7a4186d6-abb2-43d6-ab0a-9cc9f945db9b.png)

### 2.1. Relationship between ERF and depth
![image](https://user-images.githubusercontent.com/49643709/136805579-595cf66e-de52-4075-9ec1-449870a5ce8f.png)

Effective receptive field는 layer depth가 `N`일 때, `sqrt(N)`에 비례함

### 2.2. Relationship between ERF and activation function
![image](https://user-images.githubusercontent.com/49643709/136804663-dd934042-4465-43d3-ac8d-65d3958f2bde.png)

ReLU를 사용하게 되면 zero output을 가지기 쉽기에 receptive field가 output에 영향을 주지 못하는 경우가 많이 발생함

이로 인해 다른 activation function과 달리 effective receptive field가 상대적으로 Gaussian 같지 않음

### 2.3. Relationship between ERF and modules
![image](https://user-images.githubusercontent.com/49643709/136807064-0e3795ec-d47a-4b5f-99bf-a5bbd7ad553a.png)
![image](https://user-images.githubusercontent.com/49643709/136806951-e85ce0b3-0e12-4f9a-b42e-fdd40536cd1d.png)

- Dropout : no effect on ERF
- Subsampling : increase ERF
- Dilated convolution : increase ERF
- Skip-connection : decrease ERF

여기서 주목할만한 점은 dilation을 적용하면 effective receptive field가 Gaussian같지 않고 rectangular한 shape를 보여줌

### 2.4. Relationship between ERF and training
![image](https://user-images.githubusercontent.com/49643709/136810898-2d936f54-b47a-4f89-b853-7195a0758586.png)

Train을 진행하면 effective receptive field가 커짐

하지만 이를 통해 theoretical receptive field와 비슷해지지는 않음
(Pooling, downsampling이 없는 figure의 ResNet의 TCF는 74 * 74이나 ERF는 32 * 32도 커버 못하고 있으며, CamVid의 TRF는 505 * 505이나 ERF는 150 * 150 정도임)

### 3.1. New initialization

Convolution filter의 center weight는 small scale로, outside weight는 large scale로 random weight initialization

해당 initialization을 통해 he, xavier initialization보다 종종 30% 정도 더 빨리 수렴할 수 있었으나, 항상 이와 같지 않았음
(overall the benefit of this method is not always significant)

그리고 해당 방법을 사용해도 ERF는 여전히 Gaussian distibuted됨

### 3.2. Architectural changes

Local rectangular convolution window를 사용하지 말고 sparsely connect한 connection을 사용하는 모듈을 이용하여 CNN을 설계
(sparsely connect each unit to a larger area in the lower layer using the same number of connections)

Dilated convolution이 예시가 될 수 있으나 grid-like structure이기에 보다 not grid-like한 sparse connection을 사용하면 더 좋을 것이라 생각함

## 4. Etc
그외 논문에서 재미있는 문장들을 가져옴

> - Effective receptive field in deep CNNs actually grows a lot slower than we used to think and this indicates that a lot of local information is still preserved even after many convolution layers.
> -  If the ERF is smaller than the RF, this suggests that representations may retain position information, and also raises an interesting question concerning changes in the size of these fields during development.
